source/common/memory/utils.cc

```diff
@@ -13,5 +16,26 @@ void Utils::releaseFreeMemory() {
 #endif
 }
 
+/*
+  The purpose of this function is to release the cache introduced by tcmalloc,
+  mainly in xDS config updates, admin handler, and so on. all work on the main thread,
+  so the overall impact on performance is small.
+  Ref: https://github.com/envoyproxy/envoy/pull/9471#discussion_r363825985
+*/
+void Utils::tryShrinkHeap() {
+#ifdef TCMALLOC
+  // TODO(zyfjeff): Make max unfreed memory byte configurable
+  static const uint64_t MAX_UNFREED_MEMORY_BYTE = 100 * 1024 * 1024;
+  auto heap_size = Stats::totalCurrentlyReserved() - Stats::totalPageHeapUnmapped();
```

Comment:
Maybe for slightly more clarify call it `committed_heap_size`? I.e. heap size that has a physical page mapped. Also can you add a comment why you want to exclude unmapped pages from this computation, please?

Reply:
Yes, it's a little confusing here, so I'll change it to 'generic.total_physical_bytes'

https://github.com/gperftools/gperftools/blob/b85652bf268d061a0ad0aec90f0d3c29aa80a6b5/src/gperftools/malloc_extension.h#L171

https://github.com/gperftools/gperftools/blob/b85652bf268d061a0ad0aec90f0d3c29aa80a6b5/src/gperftools/malloc_extension.h#L171